lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -312,363 +314,372 @@ def concatenate_cubes(
-            blending or None.
-        blend_coord (str):
-            Name of coordinate over which merged cube is to be blended, or
-            None. If "time", triggers bounds range checks to avoid passing
-            eg mismatched accumulation periods into blending.
+    def _equalise_cubes(self, cubelist):
+        """
+        Function to equalise cubes where they do not match.
 
-    Returns:
-        result (Iris cube):
-            Merged cube.
+        Args:
+            cubelist (iris.cube.CubeList):
+                List of cubes to check and equalise.
 
-    """
-    if isinstance(cubes, iris.cube.Cube):
-        cubes = iris.cube.CubeList([cubes])
+        Returns:
+            cubelist (iris.cube.CubeList):
+                List of cubes with revised cubes. The number of cubes in this
+                list may be greater than the original number of cubes if they
+                have been sliced over mismatching dimension coordinates.
+        """
+        equalise_cube_attributes(cubelist, silent=self.silent_attributes)
+        strip_var_names(cubelist)
 
-    cubelist = _equalise_cubes(
-        cubes, model_id_attr=model_id_attr, merging=True)
+        cubelist = self._equalise_cube_coords(cubelist)
+        cubelist = self._equalise_cell_methods(cubelist)
+        for cube in cubelist:
+            check_cube_not_float64(cube, fix=True)  # TODO why here?
+        return cubelist
 
-    for i, cube in enumerate(cubelist):
-        cubelist[i] = iris.util.squeeze(cube)
+    def _equalise_cube_coords(self, cubes):
+        """
+        Function to equalise coordinates that do not match.
 
-    result = cubelist.merge_cube()
+        Args:
+            cubes (iris.cube.CubeList):
+                List of cubes to check the coords and revise.
 
-    if blend_coord is not None and blend_coord == "time":
-        # If bounds ranges did not match, "result" will not have a name
-        # on the leading dimension, but will have time and fp as aux
-        # coords.  Checking bounds ranges here raises a useful error,
-        # rather than encountering a missing blend coord downstream.
-        check_bounds_range_coords = ["time", "forecast_period"]
-        _check_bounds_ranges(result, check_bounds_range_coords)
+        Returns:
+            cubelist (iris.cube.CubeList):
+                List of cubes with revised coords. The number of cubes in this
+                list may be greater than the original number of cubes if they
+                have been sliced over mismatching dimension coordinates.
 
-    return result
+        Raises:
+            ValueError:
+                If coordinates in error_keys do not match.
+        """
+        # check for unmatching coords (returns a list of dictionaries, each
+        # with an entry for each mismatched coord.  If all coords match,
+        # returns a list of empty dictionaries.)
+        unmatching_coords = compare_coords(cubes)
```

Comment:
I wonder whether `compare_coords` would benefit from the same attention as `compare_attributes`, perhaps as part of the same Jira follow-on ticket?

Reply:
Yes, potentially a follow-on ticket to think about what the `compare_coords` function is returning and whether this can be rationalised, would sound reasonable.

Reply:
Would you be happy to create a ticket for this?